src/three/PNTSLoader.js

```diff
@@ -12,68 +23,115 @@ export class PNTSLoader extends PNTSLoaderBase {
 
 	parse( buffer ) {
 
-		return super
-			.parse( buffer )
-			.then( result => {
+		return super.parse( buffer ).then( async ( result ) => {
 
-				const { featureTable } = result;
+			const { featureTable } = result;
 
-				const POINTS_LENGTH = featureTable.getData( 'POINTS_LENGTH' );
-				const POSITION = featureTable.getData( 'POSITION', POINTS_LENGTH, 'FLOAT', 'VEC3' );
-				const RGB = featureTable.getData( 'RGB', POINTS_LENGTH, 'UNSIGNED_BYTE', 'VEC3' );
+			const material = new PointsMaterial();
+			const extensions = featureTable.header.extensions;
+			let geometry;
+
+			// handle loading the draco data
+			if ( extensions && extensions[ '3DTILES_draco_point_compression' ] ) {
+
+				const { byteOffset, byteLength, properties } = extensions[ '3DTILES_draco_point_compression' ];
+				const dracoLoader = this.manager.getHandler( 'draco.drc' );
+				if ( dracoLoader == null ) {
+
+					throw new Error( 'PNTSLoader: dracoLoader not available.' );
+
+				}
 
-				[
-					'QUANTIZED_VOLUME_OFFSET',
-					'QUANTIZED_VOLUME_SCALE',
-					'CONSTANT_RGBA',
-					'BATCH_LENGTH',
-					'POSITION_QUANTIZED',
-					'RGBA',
-					'RGB565',
-					'NORMAL',
-					'NORMAL_OCT16P',
-				].forEach( feature => {
+				// map PNTS keys to draco types
+				const attributeIDs = {};
+				for ( const key in properties ) {
 
-					if ( feature in featureTable.header ) {
+					const mappedKey = DRACO_ATTRIBUTE_MAP[ key ];
+					if ( mappedKey in attributeIds ) {
```

Comment:
I think this should be 
```
if ( ! ( mappedKey in attributeIDs ) )
```

Reply:
Oh oops - you're right this isn't correct. But I think it should be the following to make sure both fields are defined:

```js
if ( key in DRACO_ATTRIBUTE_MAP && key in properties ) {

  // ...

} 
```